### PR TITLE
test: Adapt `test_env_update_pypi_with_conda_forge`

### PR DESCRIPTION
**Pull request by jjerphan**
Resolve CI failures observed on #3450 and #3445.

### CODE REVIEW

File: micromamba/tests/test_env.py [hunk 1]
```diff
@@ -362,4 +362,4 @@ def test_env_update_pypi_with_conda_forge(tmp_home, tmp_root_prefix, tmp_path):
     pip_list_output = helpers.umamba_run("-p", env_prefix, "pip", "list", "--format=json")
     pip_packages_list = yaml.safe_load(pip_list_output)
 
-    assert any(pkg["name"] == "numpy" and pkg["version"] == "1.26.4" for pkg in pip_packages_list)
+    assert any(pkg["name"] == "numpy" for pkg in pip_packages_list)
```
Comment (Hind-M) [inline micromamba/tests/test_env.py:365]: What happens exactly? Is it updated with another version?

Comment (jjerphan) [reply micromamba/tests/test_env.py:365]: I cannot reproduce the error locally, and I needed to have a first relaxation of the test on the CI while working #3456 (hence the minimalist state of this PR).

The test suite does not fail anymore on `main`, so I am clueless for now. :shrug:

**Comment by jjerphan** [comment]
Closing since the error is not observed anymore.

**Comment by jjerphan** [comment]
Reopening after the issue was observed again.

**Comment by Hind-M** [comment]
Superseded by #3459

**Pull request closed without merging**